apps/src/propTypes.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Code.org custom react proptypes.
+ */
+import React from 'react';
+
+/**
+ * A custom React PropType to ensure that the specified
+ * component types are given in the specified order
+ * as children of the component that this prop type is used on.
+ *
+ * Example:
+ *   propTypes: {
+ *     children: childrenOfType([Heading, Body])
+ *   }
+ *
+ * In this example, the prop type does not validate if there are more than
+ * one <Heading> components in the children or if <Heading> comes after <Body>.
+ * However, the prop type will be valid if you omit <Heading> and/or <Body>.
+ *
+ * @param ...validChildrenTypes array<constructor> - a list of types to allow as children
+ */
+export function childrenOfType(...validChildrenTypes) {
```

Comment:
Even reading your tests, I have a hard time parsing what this function does.  Basically, any ordered subset of the list your provide is valid?

So, for `childrenOfType([Foo, Bar, Baz])` the following are allowed:

```
undefined
Foo
Bar
Baz
Foo, Bar
Foo, Baz
Bar, Baz
Foo, Bar, Baz
```

But we disallow any elements out of order, extra elements, or types not listed:

```
Foo, Baz, Bar
Foo, Foo, Bar, Baz
Foo, Bar, Baz, Baz
Foo, Bar, Baz, Other
```

Maybe... maybe this validator would just read better without the `forEach` method, since we're stepping through two lists side-by-side.